README.md

```diff
@@ -0,0 +1,241 @@
+Clone:
+```
+cd (your installation directory)
+git clone https://github.com/kaguyabot/Kaguya.git
+cd Kaguya
+```
+
+Update to the latest version of the source code:
+```
+git pull
+```
+
+## Building
+Kaguya requires several steps in order to actually get going, all involving a MySQL database.
+
+Create a MySQL Database
+```
+$ mysql -u root -p
+mysql> CREATE DATABASE kaguya;
+```
+
+Verify successful database creation:
+```
+mysql> SHOW DATABASES;
+
+(output)
++--------------------+
+| Database           |
++--------------------+
+| information_schema |
+| kaguya             | (<-- this line should be present)
+| mysql              |
+| performance_schema |
++--------------------+
+4 rows in set (0.00 sec)
+```
+
+Navigate to your installation folder and find the `appsettings.Development.json` file.
+You can do these steps through the command line or through a text editor.
+
+Navigate to the project directory.
+```
+$ cd (installation_directory)/Kaguya/Kaguya/Kaguya
+$ ls
+
+(output)
+appsettings.Development.json  Database/  Global.cs        Migrations/  Program.cs   Startup.cs
+appsettings.json              Discord/   Kaguya.csproj    obj/         Properties/  Web/
+bin/
+```
+
+Copy the contents of the `appsettings.json` file to `appsettings.Development.json`.
+```
+$ cp appsettings.json appsettings.Development.json
+```
+
+Modify `appsettings.Development.json` to include the following:
+- BotToken
+- OwnerId (your discord user ID)
+- Database connection string
+
+
+Open the file in vim, or edit in a text editor (VS Code).
+```bash
+$ vi -v appsettings.Development.json
+or
+$ code appsettings.Development.json
+```
+Your file should now look something like this (note that the "AdminSettings" section will change overtime, this is just an example):
+
+```json
+{
+  "Logging": {
+    "LogLevel": {
+      "Default": "Information",
+      "Microsoft": "Warning",
+      "Microsoft.Hosting.Lifetime": "Information"
+    }
+  },
+  "AllowedHosts": "*",
+  "AdminSettings": {
+    "OwnerId": YOUR_ID, // Optional
+    "OsuApiKey": "" // Optional
+  },
+  "DiscordSettings": {
+    "BotToken": "YOUR_TOKEN",
+    "MessageCacheSize": 50,
+    "AlwaysDownloadUsers": true
+  },
+  "ConnectionStrings": {
+    "Database": "server=localhost;user=root;password=YOUR_PASSWORD;database=kaguya"
+  }
+}
+```
+
+- Save this file and close it.
+
+Now, we need to use entity framework to setup the database for us.
+
+Navigate to the folder containing `Kaguya.csproj` if you have not already done so. Note that this is the same folder as the previous step.
+
+Run:
+```
+$ dotnet ef database update
+
+(output)
+Build started...
+Build succeeded.
+Done.
+```
+
+You are now able to run Kaguya.
+```
+$ dotnet run
+```
+
+## Contributing
+If you wish to contribute to the codebase, please note the following:
+- Create a fork of the code.
+- All commands must adhere to the guidelines below.
+- All changes must be made to the `v4-open-beta` branch. This is the latest stable release.
```

Comment:
@BitMasher Please verify that these steps are what actually take place when an outside contributor wants to submit changes to this repo.

Reply:
i don't think the branch on their end matters, they can create a PR from their repo into the kaguya repo, they just need to PR into the correct branch